src/LLVM_Output.cpp

```diff
@@ -313,19 +313,38 @@ std::unique_ptr<llvm::raw_fd_ostream> make_raw_fd_ostream(const std::string &fil
     return raw_out;
 }
 
+namespace {
+
+// llvm::CloneModule has issues with debug info. As a workaround,
+// serialize it to bitcode in memory, and then parse the bitcode back in.
+std::unique_ptr<llvm::Module> clone_module(const llvm::Module &module_in) {
+    // Write the module to a buffer.
+    llvm::SmallVector<char, 1024> clone_buffer;
```

Comment:
Is the 1k number based on something? This will allocate that much space on the stack and looking at the tmp build folder, I don't see any .bc files less than 1k. So I'm thinking there's potential here that we're just burning a kilobyte of stack that doesn't get used. (The 1k is probably also going to get memcpy'ed when it moves to the heap allocation but I guess it has to realloc the heap part too.) Since there's no recursion, we could bump it up to 4k or 8k, or we could drop it down to something small since it will have to use heap anyway. (Not sure if llvm has some sort of non-contiguous data structure for this. That would probably be better if so, but that's more work than tuning the number if it is doable at all.)

Reply:
It was arbitrary, I was thinking I'd just try to handle the case where the module was completely empty, but it really doesn't matter.